src/zope/server/ftp/publisher.py

```diff
@@ -41,33 +41,24 @@ def type(self, path):
 
         return self._execute(path, 'type')
 
-    def names(self, path, filter=None):
-        return self._execute(path, 'names', split=False, filter=filter)
-
-    def ls(self, path, filter=None):
-        return self._execute(path, 'ls', split=False, filter=filter)
 
     def readfile(self, path, outstream, start=0, end=None):
         return self._execute(path, 'readfile',
                              outstream=outstream, start=start, end=end)
 
-    def lsinfo(self, path):
-        return self._execute(path, 'lsinfo')
-
-    def mtime(self, path):
-        return self._execute(path, 'mtime')
-
-    def size(self, path):
-        return self._execute(path, 'size')
-
-    def mkdir(self, path):
-        return self._execute(path, 'mkdir')
-
-    def remove(self, path):
-        return self._execute(path, 'remove')
-
-    def rmdir(self, path):
-        return self._execute(path, 'rmdir')
+    _name = None
+    for _name in ('names', 'ls'):
+        f = locals()[_name] = lambda self, path, filter=None, _name=_name: self._execute(
+            path,
+            _name,
+            split=False,
+            filter=filter)
+        f.__name__ = _name
+
+    for _name in ('lsinfo', 'mtime', 'size', 'mkdir', 'remove', 'rmdir'):
+        f = locals()[_name] = lambda self, path, _name=_name: self._execute(path, _name)
+        f.__name__ = _name
+    del _name
```

Comment:
I find the old repetetive code much more readable.

Reply:
So the issue with the old code was that the methods were all identical, except for one embedded string constant. Worse, only half of them were actually called. I wasn't up to writing meaningful tests for the half that weren't called, and I really didn't like the repetition.

One other option would be to do this dynamically in `__getattr__`, but that would break introspection of the class. Another option would be to do it after the class body, which is more verbose, but eliminates the use of `locals`, which I know some people find objectionable (even though this same pattern is found in the stdlib).